app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailViewModel.kt

```diff
@@ -1,20 +1,29 @@
 package com.andrew.tasky.presentation.event_detail
 
 import android.net.Uri
-import androidx.lifecycle.SavedStateHandle
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.*
 import com.andrew.tasky.domain.AgendaItem
 import com.andrew.tasky.domain.Attendee
-import com.andrew.tasky.util.ReminderTimes
+import com.andrew.tasky.domain.repository.AgendaItemRepository
+import com.andrew.tasky.util.AgendaItemType
+import com.andrew.tasky.util.ReminderTime
+import dagger.hilt.android.lifecycle.HiltViewModel
 import java.time.LocalDate
+import java.time.LocalDateTime
 import java.time.LocalTime
+import javax.inject.Inject
 import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
 
-class EventDetailViewModel(
-    private val savedStateHandle: SavedStateHandle
+@HiltViewModel
+class EventDetailViewModel @Inject constructor(
+    private val savedStateHandle: SavedStateHandle,
+    private val repository: AgendaItemRepository
 ) : ViewModel() {
 
+    private var id: Int? = null
```

Comment:
Could directly be set to `savedStateHandle.get<String>("your_id_key")` if I get the purpose right here (the ID you pass as nav arg to this screen)